packages/number-field/src/vaadin-number-field-mixin.js

```diff
@@ -9,6 +9,12 @@ import { InputController } from '@vaadin/field-base/src/input-controller.js';
 import { InputFieldMixin } from '@vaadin/field-base/src/input-field-mixin.js';
 import { LabelledInputController } from '@vaadin/field-base/src/labelled-input-controller.js';
 
+// [type=number] inputs return an empty string for invalid numbers,
+// which makes them indistinguishable from empty values. This string
+// is used in _inputElementValue as a marker to help Flow detect and
+// clear unparsable values (if needed).
+const BAD_INPUT_STRING = 'NaN';
```

Comment:
I wonder if we could use native [`valueAsNumber`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/valueAsNumber) for number inputs which actually returns `NaN`?

Reply:
Unfortunately, `valueAsNumber` returns `NaN` also for empty values.

Reply:
Thanks, now I see. That's unfortunate.